metabase_api/metabase_api.py

```diff
@@ -778,8 +778,64 @@ def copy_collection(self, source_collection_name=None, source_collection_id=None
         self.copy_pulse(source_pulse_id=pulse_id,
                         destination_collection_id=destination_parent_collection_id,
                         destination_pulse_name=destination_pulse_name)
-  
-  
+
+
+  def search(self, q, item_type=None, archived=False, single_ret=False):
+    """
+    Search content and filter item type. Note: endpoint /api/search fall behind with web app version.
+
+    Keyword arguments:
+    q -- search input
+    item_type -- result type (default None means all type)
+    archived -- False or True
+    """
+    assert item_type in [None, 'card', 'dashboard', 'collection']
+    assert archived in [True, False]
+    ret = self.get(endpoint='/api/search', params={'q': q, 'archived': archived})
+    if item_type is not None:
+      ret = [item for item in ret if item['model'] == item_type]
+    if single_ret:
+      if len(ret) > 1:
+        raise ValueError("There is more than one item with the name '{}'".format(q))
+      else:
+        ret = ret[0]
+    return ret
+
+
+  def query(self, card_name=None, card_id=None, collection_id=None, collection_name=None, export_format=None):
+    """
+    Get saved question data.
+
+    Keyword arguments:
+    card_name -- card_name
+    card_id -- card_id, int type
+    collection_id -- collection_id
+    collection_name -- collection_name
+    """
+    if export_format not in [None, 'csv', 'json', 'xlsx']:
+      raise ValueError('{} format not supported. Supported format are csv, json, xlsx'.format(export_format))
+    if [card_id, card_name].count(None) == 2:
+      raise ValueError('Either card_id or card_name should be available')
+    if card_id is None:
+      if collection_id is None and collection_name is None:
```

Comment:
- Why not using `get_item_id` for this case as well? Is there a need for the `search` function that you defined?